### ISSUE TITLE
It starts, it closes and then it opens a new Java icon in the Dock

### ISSUE THREAD

**Issue by Michelasso** [tofi86/universalJavaApplicationStub#21]
I made a stub for the Sony Theme Creator, modifying the Info.plist file from JD-GUI and updating the universalJavaApplicationStub script to yours. the app shows its nice icon and new name in Finder, I could move it to the Dock, and it does start the Java application. The problem is that after showing the splash screen the nice icon in the Dock closes and a new one opens in the Dock's far right. This is what basically gets called:

'/Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home//bin/java' -cp '/Applications/Theme Creator.app/Contents/Resources/Java/theme-creator-v0_7_0-BETA.jar' '-splash:/Applications/Theme Creator.app/Contents/Resources/stoffa_arlecchino.png' '-Xdock:icon=/Applications/Theme Creator.app/Contents/Resources/themes.icns' '-Xdock:name=Theme Creator' -Dapple.laf.useScreenMenuBar=true se.blogspot.qdolan.loader.JarRunner

"se.blogspot.qdolan.loader.JarRunner" is the MainClass, which (obviously) is a jar runner.

More than a bug report this is a cry for help. I have been wasting 3 or 4 days on this already , I actually made it working fine once (heavily changing the script), but then I messed up and there I was again. If you want to test it yourself this is the link to the application:

https://developer.sony.com/downloads/tool/theme-creator/

PS: I  tried to attach the Info.plist as well as a text file but I am not allowed.

**Comment by tofi86** [comment]
Can you paste the Info.plist content here as plain-text?

**Comment by Michelasso** [comment]
Sure. I'd just like to let you know that it works wonder with another couple of different .jar files I tested. But I'm not a Java expert, so I really don't know what's going on.

<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>CFBundleDevelopmentRegion</key>
	<string>English</string>
	<key>CFBundleExecutable</key>
	<string>universalJavaApplicationStub.sh</string>
	<key>CFBundleGetInfoString</key>
	<string>Sony Theme Creator v0.7.0 Beta </string>
	<key>CFBundleIconFile</key>
	<string>themes.icns</string>
	<key>CFBundleIdentifier</key>
	<string>theme-creator</string>
	<key>CFBundleInfoDictionaryVersion</key>
	<string>6.0</string>
	<key>CFBundleLongVersionString</key>
	<string>0.7.0, Copyright 2015 Sony Mobile</string>
	<key>CFBundleName</key>
	<string>Theme Creator</string>
	<key>CFBundlePackageType</key>
	<string>APPL</string>
	<key>CFBundleShortVersionString</key>
	<string>0.7.0</string>
	<key>CSResourcesFileMapped</key>
	<true/>
	<key>JavaX</key>
	<dict>
		<key>ClassPath</key>
		<string>$JAVAROOT/theme-creator-v0_7_0-BETA.jar</string>
		<key>JVMVersion</key>
		<string>1.7+</string>
		<key>MainClass</key>
		<string>se.blogspot.qdolan.loader.JarRunner</string>
		<key>Properties</key>
		<dict>
			<key>apple.laf.useScreenMenuBar</key>
			<string>true</string>
		</dict>
		<key>SplashFile</key>
		<string>stoffa_arlecchino.png</string>
		<key>WorkingDirectory</key>
		<string>$JAVAROOT</string>
	</dict>
	<key>LSRequiresCarbon</key>
	<true/>
	<key>NSHighResolutionCapable</key>
	<true/>
	<key>NSHumanReadableCopyright</key>
	<string>Modified by Michele Dall'Agata</string>
	<key>NSPrincipalClass</key>
	<string>NSApplication</string>
</dict>
</plist>

**Comment by tofi86** [comment]
I'm sorry I cannot reproduce your issue! Works as expected on my Mac...

The theme app starts and it is run as expected in the first thread forever... No closing and respawning a new process...

This was the commandline the stub produced:
`"/Library/Java/JavaVirtualMachines/jdk1.8.0_60.jdk/Contents/Home/bin/java" -cp "/Users/tobiasfischer/Entwicklung/github/universalJavaApplicationStub/test/Sony-Themes.app/Contents/Resources/Java/theme-creator-v0_7_0-BETA.jar" -Xdock:icon="/Users/tobiasfischer/Entwicklung/github/universalJavaApplicationStub/test/Sony-Themes.app/Contents/Resources/paginaEPUBChecker_512.icns" -Xdock:name="Theme Creator" -Dapple.laf.useScreenMenuBar=true se.blogspot.qdolan.loader.JarRunner`


![bildschirmfoto 2015-10-29 um 00 01 52](https://cloud.githubusercontent.com/assets/365118/10805726/55bcb29c-7dd0-11e5-9bc1-c9bd3d4100c9.png)

Ignore my custom icon and Name for it...

**Comment by Michelasso** [comment]
Then I don't know what it is. I even reinstalled Java (after deleting it as in the instructions from Oracle) and also tried in my OS X Lion partition with a separate Java installation. Same identical thing.

Unless it has to do with the JDK not being installed, as instead it seem to be your case. Thanks a lot for your time anyway. I really appreciated it.

**Comment by tofi86** [comment]
yes, I ran with JDK and not JRE. but this shouldn't be the problem...
You're using the latest version of my stub script, right?

**Comment by Michelasso** [comment]
Yeah, yeah. The last version. But I have the issue also starting it from Terminal. If I don't set the splash screen option the Xdock options do nothing. I tried everything, even deleting all caches and the /var/folders/wm (I think it was wm) folder, which I have got from lsof. The mad thing is that I think I managed to make it working once. Unless after all the trials and errors now I remember badly.

It just doesn't make any sense that it respawns. Why should it?! (sorry, I'm getting very frustrated).
